cli/src/pcluster/config/config_patch.py

```diff
@@ -181,14 +181,46 @@ def _compare_list(self, base_section, target_section, param_path, data_key, fiel
         """
         update_key = field_obj.metadata.get("update_key")
 
+        # Get the lists
+        base_list = base_section.get(data_key, []) if base_section else []
+        target_list = target_section.get(data_key, []) if target_section else []
+
+        # For Tags, check if this is ONLY an order change (same content, different order)
+        if data_key == "Tags" and base_list != target_list:
+            # Check if it's an order-only change
+            try:
+                base_set = {frozenset(item.items()) for item in base_list}
```

Comment:
[Risk] What happens if the user specifies the same tag key twice in the cluster config? The list will be converted into a set, eliminating duplicates and the set comparison will not detect changes in the order. This approach would be ok only if we ensure that duplicates keys are not injected into the template